File: web/app/types/fi/oph/koski/schema/Tukijakso.ts
/**
 * Oppivelvollisen tukijaksot
 *
 * @see `fi.oph.koski.schema.Tukijakso`
 */
export type Tukijakso = {
  $class: 'fi.oph.koski.schema.Tukijakso'
  alku?: string
  loppu?: string
}

export const Tukijakso = (
  o: {
    alku?: string
    loppu?: string
  } = {}
): Tukijakso => ({ $class: 'fi.oph.koski.schema.Tukijakso', ...o })

Tukijakso.className = 'fi.oph.koski.schema.Tukijakso' as const

export const isTukijakso = (a: any): a is Tukijakso =>
  a?.$class === 'fi.oph.koski.schema.Tukijakso'

File: src/main/scala/fi/oph/koski/schema/Tukijakso.scala
package fi.oph.koski.schema

import fi.oph.scalaschema.annotation.Description

import java.time.LocalDate

@Description("Oppivelvollisen tukijaksot")
case class Tukijakso(
  @Description("Jakson alkamispäivämäärä. Muoto YYYY-MM-DD")
  alku: Option[LocalDate],
  @Description("Jakson loppumispäivämäärä. Muoto YYYY-MM-DD")
  loppu: Option[LocalDate]
) extends MahdollisestiAlkupäivällinenJakso
